main.cpp

```diff
@@ -90,8 +90,77 @@ good to go!
 
  Wait for my code review.
  */
-
 #include <iostream>
+
+struct FloatType
+{
+    float add( float lhs, float rhs )
+    {
+        return lhs + rhs;
+    }
+    float subtract( float lhs, float rhs )
+    {
+        return lhs - rhs;
+    }
+    float multiply( float lhs, float rhs )
+    {
+        return lhs * rhs;
+    }
+    float divide(float lhs, float rhs)
+    {
+        if( rhs == 0.f ) std::cout << "\nwarning, floating point division by zero returns 'inf' !\n";
+
+        return lhs / rhs;
+    }
```

Comment:
use this format:
```
if( expr )
    code
```
it's easier to read/understand.
the readability of the code should be your top priority right now.
- [ ] revise